models/iaf_cond_beta.cpp

```diff
@@ -0,0 +1,521 @@
+/*
+ *  iaf_cond_beta.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+#include "iaf_cond_beta.h"
+
+#ifdef HAVE_GSL
+
+// C++ includes:
+#include <cstdio>
+#include <iomanip>
+#include <iostream>
+#include <limits>
+
+// Includes from libnestutil:
+#include "numerics.h"
+
+// Includes from nestkernel:
+#include "exceptions.h"
+#include "kernel_manager.h"
+#include "universal_data_logger_impl.h"
+
+// Includes from sli:
+#include "dict.h"
+#include "dictutils.h"
+#include "doubledatum.h"
+#include "integerdatum.h"
+
+/* ----------------------------------------------------------------
+ * Recordables map
+ * ---------------------------------------------------------------- */
+
+nest::RecordablesMap< nest::iaf_cond_beta >
+  nest::iaf_cond_beta::recordablesMap_;
+
+namespace nest // template specialization must be placed in namespace
+{
+/*
+ * Override the create() method with one call to RecordablesMap::insert_()
+ * for each quantity to be recorded.
+ */
+template <>
+void
+RecordablesMap< iaf_cond_beta >::create()
+{
+  // use standard names whereever you can for consistency!
+  insert_(
+    names::V_m, &iaf_cond_beta::get_y_elem_< iaf_cond_beta::State_::V_M > );
+  insert_(
+    names::g_ex, &iaf_cond_beta::get_y_elem_< iaf_cond_beta::State_::G_EXC > );
+  insert_(
+    names::g_in, &iaf_cond_beta::get_y_elem_< iaf_cond_beta::State_::G_INH > );
+
+  insert_( names::t_ref_remaining, &iaf_cond_beta::get_r_ );
+}
+}
+
+/* ----------------------------------------------------------------
+ * Iteration function
+ * ---------------------------------------------------------------- */
+
+extern "C" inline int
+nest::iaf_cond_beta_dynamics( double,
+  const double y[],
+  double f[],
+  void* pnode )
+{
+  // a shorthand
+  typedef nest::iaf_cond_beta::State_ S;
+
+  // get access to node so we can almost work as in a member function
+  assert( pnode );
+  const nest::iaf_cond_beta& node =
+    *( reinterpret_cast< nest::iaf_cond_beta* >( pnode ) );
+
+  // y[] here is---and must be---the state vector supplied by the integrator,
+  // not the state vector in the node, node.S_.y[].
+
+  // The following code is verbose for the sake of clarity. We assume that a
+  // good compiler will optimize the verbosity away ...
+  const double I_syn_exc =
```

Comment:
since they are used several time, I would consider declaring the variables in ``const double &`` to make code more readable

```suggestion
  const double& V    = y[ S::V_M ];
  const double& g_ex = y[ S::G_EXC ];
  const double& g_in = y[ S::G_INH ];

  const double I_syn_exc =
```